src/vma/proto/route_table_mgr.cpp

```diff
@@ -289,7 +288,31 @@ void route_table_mgr::parse_attr(struct rtattr *rt_attribute, route_val *p_val)
 	case RTA_PREFSRC:
 		p_val->set_src_addr(*(in_addr_t *)RTA_DATA(rt_attribute));
 		break;
+	case RTA_METRICS: {
+		struct rtattr *rta = (struct rtattr *)RTA_DATA(rt_attribute);
+		int len = RTA_PAYLOAD(rt_attribute);
+		uint16_t type;
+		while (RTA_OK(rta, len)) {
+			type = rta->rta_type;
+			switch (type) {
+			case RTAX_MTU:
+				p_val->set_mtu(*(uint32_t *)RTA_DATA(rta));
+				break;
+			case RTAX_LOCK:
+				p_val->set_rt_lock(*(uint32_t *)RTA_DATA(rta));
+				break;
+			default:
+				rt_mgr_logdbg("got unexpected METRICS %d %x",
+					type, *(uint32_t *)RTA_DATA(rta));
+				break;
+			}
+			rta = RTA_NEXT(rta, len);
+		}
+	}
+	break;
```

Comment:
use the same style as https://github.com/Mellanox/libvma/blob/master/src/vma/dev/cq_mgr_mlx5.cpp#L207